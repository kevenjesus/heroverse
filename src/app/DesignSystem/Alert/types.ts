import { ReactNode } from "react"

export interface WrapperTypes {
    type: 'error' | 'success'
}

export interface AlertProps extends WrapperTypes {
    children: ReactNode
}