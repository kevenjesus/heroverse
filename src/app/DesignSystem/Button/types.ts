import { VariantsButton } from "@/app/styles/types";
import { ReactNode } from "react";

export interface ButtonProps {
    children: ReactNode
    loading?: boolean
    onClick?: React.MouseEventHandler<HTMLButtonElement>
    variant: VariantsButton
}