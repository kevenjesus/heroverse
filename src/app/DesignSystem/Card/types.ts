import { colorsType } from "@/app/styles/types"

export interface CardProps {
    title: string
    category: {
        id: number
        label: string
        variant: colorsType
    }
    actions?: {
        onEdit: () => void
        onDelete: () => void
    }
}