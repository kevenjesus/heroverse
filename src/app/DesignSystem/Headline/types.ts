import { fontSizes } from "@/app/styles/guide"
import { HeadlineType, colorsType, fontSizeType, weightTye } from "@/app/styles/types"
import { ReactNode } from "react"

export const HeadlineTypes = {
    h1: fontSizes.heading1,
    h2: fontSizes.heading2,
    h3: fontSizes.heading3,
    h4: fontSizes.heading4
}

export interface HeadlineProps {
    children: ReactNode
    size?: fontSizeType
    weight?: weightTye
    color?: colorsType
    as: HeadlineType
}