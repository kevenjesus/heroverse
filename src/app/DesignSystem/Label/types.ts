import { colorsType } from "@/app/styles/types";
import { ReactNode } from "react";


export interface LabelItemProps {
    variant: colorsType
}

export interface LabelProps extends LabelItemProps {
    children: ReactNode
}

