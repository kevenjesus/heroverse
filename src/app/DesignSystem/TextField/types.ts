import { ChangeEvent, EventHandler } from "react"

export interface TextFieldInputProps {
    name: string
    onChange: (event: ChangeEvent<HTMLInputElement>) => void
    value: string
    type: string
    placeholder?: string
}