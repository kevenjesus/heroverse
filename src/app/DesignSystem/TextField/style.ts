import { colors, fontSizes } from '@/app/styles/guide'
import styled from 'styled-components'

export const TextFieldInput = styled.input`
    width: 100%;
    height: 45px;
    background: #fff;
    border: 2px solid ${colors.black};
    padding: 0 20px;
    font-size: ${fontSizes.regular};
    color: ${colors.black};
    display: flex;
    align-items: center;
`