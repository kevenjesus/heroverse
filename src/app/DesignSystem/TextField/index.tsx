import Label from '../Label'
import * as S from './style'
import { TextFieldInputProps } from './types'

export default function TextField(props: TextFieldInputProps) {
    return (
        <S.TextFieldInput role="textbox" {...props} />
    )
}