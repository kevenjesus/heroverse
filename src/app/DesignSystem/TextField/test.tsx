import { fireEvent, render, screen } from '@testing-library/react';
import TextField from '.';
import { TextFieldInputProps } from './types';
import '@testing-library/jest-dom';
import { createEvent } from '@testing-library/dom';

describe('TextField', () => {
  const onChangeMock = jest.fn();
  const props: TextFieldInputProps = {
    name: 'test-field',
    onChange: onChangeMock,
    value: '',
    type: 'text',
    placeholder: 'Enter your text here',
  };

  it('should render the component', () => {
    const { container } = render(<TextField {...props} />);
    expect(container).toMatchSnapshot();
  });

  
  it('should call onChange when input value changes', () => {
    const onChangeMock = jest.fn();
    render(<TextField type="text" value="test" name="test-field" placeholder="Enter your text here" onChange={onChangeMock} />);
    const input = screen.getByRole('textbox');
  
    fireEvent(input, createEvent.change(input, { target: { value: 'Test Value' } }));
  
    expect(onChangeMock).toHaveBeenCalled()
  });
  

  it('should have correct placeholder attribute', () => {
    const { getByPlaceholderText } = render(<TextField {...props} />);
    const input = getByPlaceholderText('Enter your text here');

    expect(input).toBeInTheDocument();
  });
});
