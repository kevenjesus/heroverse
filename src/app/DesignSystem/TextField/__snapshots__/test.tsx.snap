// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render the component 1`] = `
<div>
  <input
    class="sc-dmqHEX aYPxC"
    name="test-field"
    placeholder="Enter your text here"
    role="textbox"
    type="text"
    value=""
  />
</div>
`;
